### PR TITLE
Fix Encoding::UndefinedConversionError in activation.rb

### PR DESCRIPTION
**Pull request by mckeed**
### Motivation

Another fix for #3200. The [last one](https://github.com/Shopify/ruby-lsp/issues/3161) only checked for malformed UTF-8, not correct UTF-8 with ASCII encoding, which is the issue.

### Implementation

I changed it so it converts everything to UTF-8 to avoid the Encoding::UndefinedConversionError error.

### Automated Tests

I updated the test so it fails on the previous fix.

### Manual Tests

On macOS, add a unicode character to your system shell prompt and launch vscode using the .app, not from the command line.

### CODE REVIEW

File: test/integration_test.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ def test_ruby_lsp_check_works
   def test_activation_script_succeeds_even_on_binary_encoding
     ENV["LC_ALL"] = "C"
     ENV["LANG"] = "C"
-    ENV["PS1"] = "\xE2".b
+    ENV["PS1"] = "\xE2\x96\xB7".b
```
Comment (vinistock) [inline test/integration_test.rb:32]: We still want to verify that we're rejecting values that cannot be converted to UTF-8. Can you please duplicate this test so that we have examples for both?

File: vscode/activation.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
-env = ENV.filter_map { |k, v| "#{k}RUBY_LSP_VS#{v}" if v.dup.force_encoding(Encoding::UTF_8).valid_encoding? }
+env = ENV.filter_map { |k, v| v = v.dup.force_encoding(Encoding::UTF_8); "#{k}RUBY_LSP_VS#{v}" if v.valid_encoding? }
```
Comment (vinistock) [inline vscode/activation.rb:1]: I guess we can simplify this and just try to encode straight away and rescue the conversion error.

```suggestion
env = ENV.filter_map do |k, v|
 "#{k}RUBY_LSP_VS#{v.encode(Encoding::UTF_8)}"
rescue Encoding::UndefinedConversionError
  nil
end
```

**Review by vinistock — COMMENTED** [review]
Thanks for the PR. Let's get this fixed

File: test/integration_test.rb (hunk 1 above)
Comment (mckeed) [reply test/integration_test.rb:32]: Done. I'm not sure how the result of this is used by the extension – do we also want this test to assert that valid unicode in ENV is successfully returned from the activation script?

Comment (vinistock) [reply test/integration_test.rb:32]: I tried creating an end to end test using the extension, but there was no easy way to get the `LANG=C` to apply to the shell invoked by NodeJS when running our activation script.

For now, it's fine to just keep these integration tests.

File: vscode/activation.rb (hunk 2 above)
Comment (vinistock) [reply vscode/activation.rb]: Interesting. I didn't expect `encode` to raise even when it's possible to force the encoding to UTF-8. You're right, let's go with the original then.

```ruby
env = ENV.filter_map do |k, v|
  utf_8_value = v.dup.force_encoding(Encoding::UTF_8)
  "#{k}RUBY_LSP_VS#{utf_8_value}" if utf_8_value.valid_encoding?
end
```
